src/main/java/com/artipie/auth/AuthFromYaml.java

```diff
@@ -52,12 +56,38 @@
      * Ctor.
      * @param cred Credentials settings
      */
-    public YamlAuth(final YamlMapping cred) {
+    public AuthFromYaml(final YamlMapping cred) {
         this.cred = cred;
     }
 
     @Override
     public Optional<String> user(final String user, final String pass) {
-        throw new NotImplementedException("Not yet implemented");
+        final YamlMapping users = this.cred.yamlMapping("credentials");
+        Optional<String> res = Optional.empty();
+        if (users != null && users.yamlMapping(user) != null) {
+            final String stored = users.yamlMapping(user).string("pass");
+            if (stored != null && check(stored, pass)) {
+                res = Optional.of(user);
+            }
+        }
+        return res;
+    }
+
+    /**
+     * Obtains password from settings by username.
+     * @param stored Password from settings
+     * @param given Password to check
+     * @return Password if present and properly formatted
+     */
+    private static boolean check(final String stored, final String given) {
```

Comment:
@olenagerasimova I think this name doesn't describe this method correctly according to javadocs